packages/@react-aria/calendar/src/useCalendarCell.ts

```diff
@@ -288,10 +288,8 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
 
       // Scroll into view if navigating with a keyboard, otherwise
       // try not to shift the view under the user's mouse/finger.
-      // Only scroll the direct scroll parent, not the whole page, so
-      // we don't scroll to the bottom when opening date picker popover.
-      if (getInteractionModality() !== 'pointer') {
-        scrollIntoView(getScrollParent(ref.current) as HTMLElement, ref.current);
+      if (getInteractionModality() === 'keyboard') {
```

Comment:
Does changing this if statement affect screen reader usage (where modality would be virtual). Also does changing it to fully scroll into view break the comment that was removed?

Reply:
Double checked via TalkBack/VoiceOver on iPad/Android/desktop, it doesn't seem to affect screen reader usage, cells still scroll into view when focused

> Also does changing it to fully scroll into view break the comment that was removed?

So `scrollIntoViewFully` won't cause scrolling on the body if `preventScroll` is active (aka overlay is open) so it doesn't break that comment.